substrate/frame/support/src/traits/try_runtime/mod.rs

```diff
@@ -226,6 +245,59 @@ impl<BlockNumber: Clone + core::fmt::Debug + AtLeast32BitUnsigned> TryState<Bloc
 
 				result
 			},
+			Select::AllExcept(ref excluded_pallet_names) => {
+				let try_state_fns: &[(
+					&'static str,
+					fn(BlockNumber, Select) -> Result<(), TryRuntimeError>,
+				)] = &[for_tuples!(
+					#( (<Tuple as crate::traits::PalletInfoAccess>::name(), Tuple::try_state) ),*
+				)];
+
+				excluded_pallet_names.iter().for_each(|excluded_name| {
```

Comment:
We could filter out here the names that should not be used with `retain` in `try_state_fns`. Then we don't need the extra iteration below.